integrations/connext/components/form-connext-xtransfer.tsx

```diff
@@ -0,0 +1,504 @@
+
+import { motion } from 'framer-motion'
+import Image from 'next/image'
+import { MdOutlineSwapHoriz } from 'react-icons/md'
+import { RxCross2 } from 'react-icons/rx'
+import { formatUnits } from 'viem'
+import { useAccount, useBalance, useNetwork, useSendTransaction, useSwitchNetwork } from 'wagmi'
+
+import WalletConnectCustom from '@/components/blockchain/wallet-connect-custom'
+import { IsWalletConnected } from '@/components/shared/is-wallet-connected'
+import { IsWalletDisconnected } from '@/components/shared/is-wallet-disconnected'
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select'
+import Spinner from '@/components/ui/spinner'
+import { Switch } from '@/components/ui/switch'
+import { FADE_DOWN_ANIMATION_VARIANTS } from '@/config/design'
+
+import { useApproveIfNeeded } from '../hooks/use-approve-if-needed'
+import { useEstimatedAmount } from '../hooks/use-estimated-amount'
+import { useEstimatedRelayerFee } from '../hooks/use-estimated-relayer-fee'
+import { useLatestTransfers } from '../hooks/use-latest-transfers'
+import { useSupportedTransfer } from '../hooks/use-supported-transfer'
+import { useXcall } from '../hooks/use-xcall'
+import { mainnetAssets, testnetAssets } from '../utils/assets'
+import { mainnetChains, testnetChains } from '../utils/chains'
+import { arbitrumDomainId, arbitrumGoerliId, optimismDomainId, optimismGoerliDomainId } from '../utils/constants'
+import { Asset } from '../utils/types'
+
+interface FormConnextXTransferProps {
+  isMainnet: boolean
+  setIsMainnet: Dispatch<SetStateAction<boolean>>
+}
+
+export function FormConnextXTransfer({ isMainnet, setIsMainnet }: FormConnextXTransferProps) {
+  const { chain } = useNetwork()
+  const { address } = useAccount()
+  const { isLoading: isSwitchingChain, switchNetwork } = useSwitchNetwork()
+
+  const [originChain, setOriginChain] = useState(optimismDomainId)
+  const [destinationChain, setDestinationChain] = useState(arbitrumDomainId)
+  const [asset, setAsset] = useState('eth')
+  const [amount, setAmount] = useState<string>('')
+  const [showTransferStatus, setShowTransferStatus] = useState(false)
+  const [contractApproved, setContractApproved] = useState(false)
+
+  const getChain = (domainId: string) => {
+    return isMainnet
+      ? mainnetChains.find((mainnetChain) => mainnetChain.domain_id == domainId)
+      : testnetChains.find((testnetChain) => testnetChain.domain_id == domainId)
+  }
+  const getAsset = (): Asset | undefined => {
+    return isMainnet
+      ? mainnetAssets.find((mainnetAsset) => mainnetAsset.id === asset)
+      : testnetAssets.find((testnetAsset) => testnetAsset.id === asset)
+  }
+  const { data: originBalance } = useBalance({
+    address,
+    token: getAsset()?.contracts.find((contract) => contract.chain_id === getChain(originChain)?.chain_id)?.contract_address as `0x${string}`,
+    chainId: getChain(originChain)?.chain_id ?? 1,
+  })
```

Comment:
Add the `watch` parameter to update the balance on new blocks, that will update the user's balance after a transaction.
```suggestion
    watch: true,
  })
```